test/test_transformations.py

```
@@ -0,0 +1,368 @@
+        ("text\twith\ttabs", "text with tabs", "Tabs converted to spaces"),
+        ("multiple   \t  spaces", "multiple spaces", "Mixed spaces normalized"),
+    ]
+
+    all_passed = True
+    for input_text, expected, description in test_cases:
+        result = normalize_whitespace(input_text)
+        if result == expected:
+            print(f"  ✓ {description}")
+        else:
+            print(f"  ✗ {description}")
+            print(f"    Expected: {repr(expected)}")
```

Ruff (0.14.3)

Use explicit conversion flag

Replace with conversion flag

(RUF010)